Formula/gitversion.rb

```diff
@@ -0,0 +1,32 @@
+class Gitversion < Formula
+  desc "Easy semantic versioning for projects using Git"
+  homepage "https://github.com/GitTools/GitVersion"
+  url "https://www.nuget.org/api/v2/package/GitVersion.CommandLine/3.6.2"
+  version "3.6.2"
+  sha256 "db05ebacfb76ddca26180f0c70060fbb89800e434d3d55b524785d0da37dbac8"
+
+  bottle :unneeded
+
+  depends_on "mono"
+
+  def install
+    libexec.install Dir["*"]
+    (bin/"gitversion").write <<-EOS.undent
+      #!/bin/sh
+      exec "#{HOMEBREW_PREFIX}/opt/mono/bin/mono" "#{libexec}/tools/GitVersion.exe" "$@"
+    EOS
+  end
+
+  test do
+    # Circumvent GitVersion's build server detection scheme:
+    ENV["JENKINS_URL"] = nil
+
+    (testpath/"test.txt").write("test")
+    system "git", "init"
+    system "git", "add", "test.txt"
+    system "git", "commit", "-q", "--author='Test <test@example.com>'", "--message='Test'"
+    output = shell_output "#{bin}/gitversion -output json", 1
+    puts output
```

Comment:
@DomT4 I added this [as suggested by @MikeMcQuaid](https://github.com/Homebrew/homebrew-core/pull/3709#issuecomment-238544756) to figure out why the test was failing. It would be neat if the default behavior for `shell_output()` was to `puts` the `stdout` of the executable if it exited with an unexpected code, but I don't think it can do that.

Either way, I plan on removing that and changing the `1` to a `0` once I can see that the build and test is executed successfully.